### PR TITLE
Migrate to NPM, Update XTerm Packages

### PR DESCRIPTION
**Pull request by crutchcorn**
This PR:

- Migrates away from Yarn in favor of NPM per convos with @Tyriar 
- Updates the deps of `xterm` -> `@xterm/xterm` namespacing issues for the website
- Updates TypeDoc to the lowest supported version required for the Gulp addon to not throw a legacy peer dep warning with new NPM
- Adds a temporary tsconfig item to the TypeDoc configuration to sidestep build errors with TS>5
- Regenerates docs to the new codebase of XTerm and TypeDoc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,6 @@ bundle exec jekyll serve
 ## Update API documentation
 
 ```bash
-yarn
+npm i
```
Comment (Tyriar) [inline README.md:41]: `npm ci`?

File: _docs/api/terminal/classes/terminal.md [hunk 2]
```diff
@@ -99,200 +105,353 @@ Name | Type | Description |
...
-*Defined in [xterm.d.ts:839](https://github.com/xtermjs/xterm.js/blob/5.3.0/typings/xterm.d.ts#L839)*
+*Defined in [xterm.d.ts:867](https://github.com/xtermjs/xterm.js/blob/5.5.0/typings/xterm.d.ts#L867)*
+
+Gets or sets the terminal options. This supports setting multiple
+options.
+
+**`example`** Get a single option
+```ts
+console.log(terminal.options.fontSize);
+```
```
Comment (Tyriar) [inline _docs/api/terminal/classes/terminal.md:336]: Love that it brings this stuff in 🎉

File: package.json [hunk 3]
```diff
@@ -16,11 +16,12 @@
     "fs-extra": "^9.0.0",
     "gulp": "^4.0.2",
     "gulp-typedoc": "^2.2.4",
-    "typedoc": "^0.15.0",
```
Comment (Tyriar) [inline package.json:19]: Need to commit the package-lock.json too, it might be .gitignored?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (crutchcorn) [reply package.json:19]: Oh! Right! Will do

File: README.md (hunk 1 above)
Comment (crutchcorn) [reply README.md:41]: Right right! Sorry, lemme fix this!

**Review by Tyriar — APPROVED** [review]
Nice work 👍

**Pull request merged**